broadinstitute/pearl
# [JN-364] fixing display of past survey responses

(the actual fix is about 10 lines of code, the remaining 1000 are test data and tests)

Previously, if an enrollee had filed out a past version of a survey, the results would not appear in the admin tool.  This makes the survey lookup a little more robust so that it matches the form by stableId rather than response id.  Note that this still is far from perfect--ideally we'd fetch the actual form they responded to so we could show the correct question text.  But that's more complicated than this is worth, given that the form could vary by answer (e.g. they might have answered some questions on version 1 of the form, and others on version 2, and others on version 3).  If their answer was to a form other than the latest, we add a version indicator next to the stableId.  
![image](https://github.com/broadinstitute/juniper/assets/2800795/c1fbb1a9-fbc4-4aee-bca5-1d01f9fd29f4)

And fixing this bug has already vastly exceeded the time I wanted to spend on it, so we'll move along for now.


TO TEST:
1. repopulate the demo portal `./scripts/populate_portal demo`
2. Go to https://localhost:3000/demo/studies/heartdemo/env/sandbox/participants/HDVERS/surveys/hd_hd_socialHealth
3. confirm you see the participant's survey answers (note that the question text to their "neighborhoodWalkable" doesn't appear, since that question was removed from the current version, again, far from perfect, but better than not showing at all), along with 'v1' indicators next to the stableId.
